modules/phase_field/src/userobjects/EBSDReader.C

```diff
@@ -156,16 +159,19 @@ EBSDReader::readFile()
     b.phi1 = b.Phi = b.phi2 = 0.0;
   }
 
-  // Iterate through data points to get average variable values for each grain
+  // Array of vectors to store quaternions of each grain
+  std::vector<std::vector<Eigen::Quaternion<Real>>> Quat(_grain_num);
+
+  // Iterate through data points to store orientation information for each grain
   for (auto & j : _data)
   {
     EBSDAvgData & a = _avg_data[_global_id_map[j._feature_id]];
-    EulerAngles & b = _avg_angles[_global_id_map[j._feature_id]];
+    EulerAngles angles;
 
-    // use Eigen::Quaternion<Real> here?
-    b.phi1 += j._phi1;
-    b.Phi += j._Phi;
-    b.phi2 += j._phi2;
+    // convert Euler angles to quaternions
+    Eigen::Quaternion<Real> q = angles.ToQuaternion(j._phi1, j._Phi, j._phi2);
```

Comment:
```suggestion
    Eigen::Quaternion<Real> q = j.toQuaternion();
```

Fix this method to use the angles of the object you are calling it on!